neutrino.go

```diff
@@ -86,6 +86,22 @@ var (
 	DefaultBlockCacheSize uint64 = 4096 * 10 * 1000 // 40 MB
 )
 
+// SideLoadOpt defines the config required to sideload headers into the DB.
+type SideLoadOpt struct {
+	// SkipVerify indicates if we are to run verification on the headers from the
+	// sideLoad source.
+	SkipVerify bool
+
+	// Path contains the location of the sideload resource.
+	Path string
+
+	// Enabled indicates if sideloading is turned on or off
+	Enabled bool
+
+	// SourceType indicates the format of a sideload source.
+	SourceType string
```

Comment:
This isn't very specified. We should either drop it, or use an enum to gain better typing.

Reply:
To create a `sourceType` with an underlying `string` type?

Reply:
Or `type SouceType uint8`

But if we go with the interface approach, then we won't need this IMO.